code/clue.js

```diff
@@ -1,59 +1,158 @@
 
 const suspects = [
   mrGreen,
+  professorPlum,
+  missScarlet,
+  colonelMustard,
+  mrsPeacock,
   mrsWhite
-  // ...  and the rest
-]
+];
 
-const weapons = []
-
-const rooms = []
-
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
+// FUNCTION RANDOMLY SELECT ONE ITEM FROM THE ARRAY 
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
 // CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
 
-// This function will be invoked when you click on the killer card.
-const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
+const mystery = {
+  suspects,
+  weapons:[],
+  rooms:[] 
+};
+
+
+// FUNCTIONS + INVOKE FUNCTIONS
+
+// FUNCTIONS pickKiller
+const pickKiller = () => {
+  mystery.killer = randomSelector(suspects);killerImage.src = mystery.killer.image;
   document.getElementById('killerCard').style.background = mystery.killer.color
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+};
+document.getElementById(
+  'killerCard'
+  ).addEventListener('click', pickKiller);
+
+
+// FUNCTIONS pickWeapon
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons)
+  document.getElementById('weaponCard').style.background = mystery.weapon.color
   document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+    'weaponName'
+).innerHTML = `${mystery.weapon}`
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+document.getElementById(
+  'weaponCard'
+  ).addEventListener('click', pickWeapon);
+
+// FUNCTIONS pickRoom
+ const pickRoom = () => {
+  mystery.room = randomSelector(rooms)
+  document.getElementById(
+    'roomName'
+  ).innerHTML = `${mystery.room}`
+ }
+
+ document.getElementById(
+  'roomCard'
+  ).addEventListener('click', pickRoom);
+
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+// FUNCTION revealMystery 
+
+const revealMystery = () => {
+  const killerAnnoncement = document.getElementById("revealButton").innerText;
+  document.getElementById("killerMessage").innerHTML = `Psst! The murder was committed by ${mrsPeacock.firstName} ${mrsPeacock.lastName}, in the ${rooms[2]} with a ${weapons[8]}.`;
```

Comment:
It seems like the answer will always be "Psst! The murder was committed by Eleanor Peacock, in the kitchen with a pistol."  

Maybe you forgot to push a commit that uses the `mystery` object values that you set in `pickRoom`, `pickWeapon` and `pickKiller`?